src/psyclone/psyir/transformations/omp_task_trans.py

```diff
@@ -0,0 +1,158 @@
+        '''
+        # Disallow CodeBlocks inside the region
+        if len(node.walk(CodeBlock)) > 0:
+            raise GenerationError(
+                "OMPTaskTransformation cannot be applied to a region "
+                "containing a code block")
+        super().validate(node, options)
+
+    def _directive(self, children, collapse=None):
+        '''
+        Creates the type of directive needed for this sub-class of
+        transformation.
+
+        :param children: list of Nodes that will be the children of \
+                         the created directive.
+        :type children: list of :py:class:`psyclone.psyir.nodes.Node`
+        :param collapse: A required parameter from parent class. Must
+                         never be set for TaskTrans (is None).
+        :type collapse: None.
+
+        :raises TransformationError: if the collapse attribute is set.
+
+        :returns: The directive created for the OpenMP Task Directive
+        :rtype: :py:class:`psyclone.psyGen.DynamicOMPTaskDirective`
+
+        '''
+        if collapse is not None:
+            raise TransformationError("Collapse attribute should not be set "
+                                      "for OMPTaskTrans")
+
+        _directive = DynamicOMPTaskDirective(children=children)
+        return _directive
+
+    def _inline_kernels(self, node):
+        '''
+        Searches the PsyIR tree inside the directive and inlines any kern
+        objects found.
+        '''
+
+        kerns = node.walk(Kern)
+        kintrans = KernelModuleInlineTrans()
+        cond_trans = FoldConditionalReturnExpressionsTrans()
+        intrans = InlineTrans()
+        for kern in kerns:
+            kintrans.apply(kern)
+            cond_trans.apply(kern.get_kernel_schedule())
+            kern.lower_to_language_level()
+
+        calls = node.walk(Call)
+        for call in calls:
+            intrans.apply(call)
+
+    def apply(self, node, options=None):
+        '''Apply the OMPTaskTrans to the specified node in a Schedule.
+
+        Can only be applied to a Loop.
+
+        The specified node is wrapped by directives during code generation
+        like so:
+
+        .. code-block:: fortran
+
+          !$OMP TASK
+          ...
+          !$OMP END TASK
+
+        At code-generation time (when
+        :py:meth:`OMPTaskDirective.gen_code` is called), this node must be
+        within (i.e. a child of) an OpenMP Serial region (OpenMP Single or
+        OpenMP Master)
+
+        Any kernels or Calls will be inlined into the region before the task
+        transformation is applied.
+
+        :param node: the supplied node to which we will apply the \
+                     OMPTaskTrans transformation
+        :type node: :py:class:`psyclone.psyir.nodes.Loop`
+        :param options: a dictionary with options for transformations\
+                        and validation.
+        :type options: dictionary of string:values or None
+        '''
+        if not options:
```

Comment:
This method must call `validate()` first.

Reply:
Added.